### ISSUE TITLE
Cask download failed

### ISSUE THREAD

**Issue by LegendNoobz** [Homebrew/homebrew-cask#141709]
### Verification

- [X] I understand that [if I ignore these instructions, my issue may be closed without review](https://github.com/Homebrew/homebrew-cask/blob/master/doc/faq/closing_issues_without_review.md).
- [X] I have retried my command with `--force`.
- [X] I ran `brew update-reset && brew update` and retried my command.
- [X] I ran `brew doctor`, fixed as many issues as possible and retried my command.
- [X] I have checked the instructions for [reporting bugs](https://github.com/Homebrew/homebrew-cask#reporting-bugs).
- [X] I made doubly sure this is not a [checksum does not match](https://docs.brew.sh/Common-Issues#cask---checksum-does-not-match) error.

### Description of issue

I tried to install metasploit with command brew install --cask metasploit. And the output like this :

==> Downloading https://osx.metasploit.com/metasploit-framework-6.3.3%2B20230213
curl: (22) The requested URL returned error: 404                              

Error: Download failed on Cask 'metasploit' with message: Download failed: https://osx.metasploit.com/metasploit-framework-6.3.3%2B20230213112950-1rapid7-1.x86_64.pkg

### Command that failed

brew install --cask metasploit

### Output of command with `--verbose --debug`

```shell
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading metasploit
==> Cask::Installer#install
==> Printing caveats
==> Cask::Installer#fetch
/usr/bin/env /opt/homebrew/Library/Homebrew/shims/shared/curl --disable --cookie /dev/null --globoff --show-error --user-agent Homebrew/4.0.3-80-g336c2c7\ \(Macintosh\;\ arm64\ Mac\ OS\ X\ 12.6\)\ curl/7.79.1 --header Accept-Language:\ en --retry 3 --location --silent --head --request GET https://osx.metasploit.com/metasploit-framework-6.3.3\%2B20230213112950-1rapid7-1.x86_64.pkg
==> Downloading https://osx.metasploit.com/metasploit-framework-6.3.3%2B20230213112950-1rapid7-1.x86_64.pkg
/usr/bin/env /opt/homebrew/Library/Homebrew/shims/shared/curl --disable --cookie /dev/null --globoff --show-error --user-agent Homebrew/4.0.3-80-g336c2c7\ \(Macintosh\;\ arm64\ Mac\ OS\ X\ 12.6\)\ curl/7.79.1 --header Accept-Language:\ en --retry 3 --location --head https://osx.metasploit.com/metasploit-framework-6.3.3\%2B20230213112950-1rapid7-1.x86_64.pkg
/usr/bin/env /opt/homebrew/Library/Homebrew/shims/shared/curl --disable --cookie /dev/null --globoff --show-error --user-agent Homebrew/4.0.3-80-g336c2c7\ \(Macintosh\;\ arm64\ Mac\ OS\ X\ 12.6\)\ curl/7.79.1 --header Accept-Language:\ en --fail --retry 3 --location --remote-time --output /Users/tiasastuti/Library/Caches/Homebrew/downloads/4219b22ce26a211c9856778214c5c91bd8bf8e988fa5118db7709d0faa45538e--metasploit-framework-6.3.3\+20230213112950-1rapid7-1.x86_64.pkg.incomplete https://osx.metasploit.com/metasploit-framework-6.3.3\%2B20230213112950-1rapid7-1.x86_64.pkg
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (22) The requested URL returned error: 404
Error: Download failed on Cask 'metasploit' with message: Download failed: https://osx.metasploit.com/metasploit-framework-6.3.3%2B20230213112950-1rapid7-1.x86_64.pkg
/opt/homebrew/Library/Homebrew/download_strategy.rb:418:in `rescue in fetch'
/opt/homebrew/Library/Homebrew/download_strategy.rb:415:in `fetch'
/opt/homebrew/Library/Homebrew/cask/download.rb:25:in `fetch'
/opt/homebrew/Library/Homebrew/cask/installer.rb:181:in `download'
/opt/homebrew/Library/Homebrew/cask/installer.rb:69:in `fetch'
/opt/homebrew/Library/Homebrew/cask/installer.rb:100:in `install'
/opt/homebrew/Library/Homebrew/cask/cmd/install.rb:110:in `block in install_casks'
/opt/homebrew/Library/Homebrew/cask/cmd/install.rb:109:in `each'
/opt/homebrew/Library/Homebrew/cask/cmd/install.rb:109:in `install_casks'
/opt/homebrew/Library/Homebrew/cmd/install.rb:185:in `install'
/opt/homebrew/Library/Homebrew/brew.rb:93:in `<main>'
```

### Output of `brew doctor` and `brew config`

```shell
Your system is ready to brew. - brew doctor

Brew config :
HOMEBREW_VERSION: 4.0.3-80-g336c2c7
ORIGIN: https://github.com/Homebrew/brew
HEAD: 336c2c792d478d5514291782467a30ebe91c8e84
Last commit: 7 hours ago
Core tap JSON: 23 Feb 06:09 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.8 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 14.0.0 build 1400
Git: 2.37.1 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.79.1 => /usr/bin/curl
macOS: 12.6-arm64
CLT: 14.2.0.0.1.1668646533
Xcode: N/A
Rosetta 2: false
```

### Output of `brew tap`

```shell
nothing output
```

**Comment by miccal** [comment]
There is an open PR to fix this: https://github.com/Homebrew/homebrew-cask/pull/141679

Unfortunately, we are having issues with merging PR's at the moment.

**Comment by LegendNoobz** [comment]
> There is an open PR to fix this: #141679
> 
> Unfortunately, we are having issues with merging PR's at the moment.

Ok, i will close this issue. Thank you for your information.
